### PR TITLE
Revert #7 and support duplicated test names in `findTest`

### PR DESCRIPTION
**Pull request by frouioui**
This PR reverts #7 and adds support for duplicated test names in the `findTest` method. This method is responsible for applying the end result of a test to the report, because we had duplicated test names the method was unable to know which test had passed or not.

### CODE REVIEW

File: parser/gotest/report_builder.go [hunk 1]
```diff
@@ -278,11 +278,18 @@ func (b *reportBuilder) findTest(name string) (int, bool) {
...
 		return b.lastID, true
 	}
+	firstMatch := -1
 	for i := b.nextID; i > 0; i-- {
 		if test, ok := b.tests[i]; ok && test.Name == name {
-			return i, true
+			firstMatch = i
+			if test.Result == gtr.Unknown {
+				return i, true
+			}
```
Comment (harshit-gangal) [inline parser/gotest/report_builder.go]: should this change not continue to look further till `test.Result != gtr.Unknown`

**Comment by harshit-gangal** [comment]
I cannot see the #7 revert changes here. 
Also, kindly add the test cases in `testdata` package.

File: parser/gotest/report_builder.go (hunk 1 above)
Comment (frouioui) [reply parser/gotest/report_builder.go]: No, we want to use results with `gtr.Unknown` as they will be set with their proper value right after the call to `findTest`.

**Review by harshit-gangal — COMMENTED** [review]
I think you should add one here for end to end report generation
`https://github.com/vitessio/go-junit-report/blob/main/internal/gojunitreport/go-junit-report_test.go` `https://github.com/vitessio/go-junit-report/tree/main/testdata`

**Comment by harshit-gangal** [comment]
> I think you should add one here for end to end report generation `https://github.com/vitessio/go-junit-report/blob/main/internal/gojunitreport/go-junit-report_test.go` `https://github.com/vitessio/go-junit-report/tree/main/testdata`

We should take the CI test output for vtadmin and generate the report and see that those tests are not reflected in the report as error.

**Comment by frouioui** [comment]
Thanks for the feedback, I have added the golden test in 1bc04d5 @harshit-gangal

**Review by harshit-gangal — APPROVED** [review]

**Pull request merged**